keras/optimizers.py

```diff
@@ -209,7 +209,7 @@ class RMSprop(Optimizer):
         - [rmsprop: Divide the gradient by a running average of its recent magnitude](http://www.cs.toronto.edu/~tijmen/csc321/slides/lecture_slides_lec6.pdf)
     """
 
-    def __init__(self, lr=0.001, rho=0.9, epsilon=1e-8, decay=0.,
+    def __init__(self, lr=0.001, rho=0.9, epsilon=K.epsilon(), decay=0.,
```

Comment:
We should not call functions in signatures:
- Changing the value of the global epsilon later will not change it in the signature
- It harms readability -- much easier to read the default value if it's explicit.

Besides, if it's in the signature it's meant to be a parameter of the class/function, independent from the global epsilon (like in BN). Same applies to all optimizers here.

Reply:
Right. Should I make `epsilon=None`, and set it to K.epsilon() in the body, like it's done elsewhere in the code (precisely because the default may change). 1e-8 is not going to work for fp16 (it's actually smaller than even minimum positive subnormal, and it's inverse is way bigger than max).

Reply:
Yes, that's reasonable. The default value should be specified in the docstring then.

I assume the optimizer behavior will not differ to any measurable extent with this change.